dali/operators/generic/roi_random_crop.cc

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <random>
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/util/batch_rng.h"
+#include "dali/pipeline/operator/arg_helper.h"
+
+namespace dali {
+
+DALI_SCHEMA(ROIRandomCrop)
+    .DocStr(R"code(Produces a fixed shape cropping window, randomly placed so that as much of the
+provided region of interest (ROI) is contained in it.
+
+If the ROI is bigger than the cropping window, the cropping window will be a subwindow of the ROI.
+If the ROI is smaller than the cropping window, the whole ROI should be contained in the cropping window.
+
+If an input shape (``in_shape``) is given, the resulting cropping window is selected to be within the
+bounds of that input shape. Alternatively, the input data subject to cropping can be fed directly to
```

Comment:
```suggestion
bounds of that input shape. Alternatively, the input data subject to cropping can be passed to the operator, in
which case the cropping window will be constrained to the shape of the data.
```
`fed` suggests that the data itself is consumed - and I think we should make it very clear that the operator doesn't touch the data.